xbmc/windowing/gbm/GBMUtils.cpp

```diff
@@ -21,18 +21,48 @@
 #include "GBMUtils.h"
 #include "utils/log.h"
 
-bool CGBMUtils::InitGbm(struct gbm *gbm, int hdisplay, int vdisplay)
+bool CGBMUtils::CreateDevice(struct gbm *gbm, int fd)
 {
-  gbm->width = hdisplay;
-  gbm->height = vdisplay;
+  if (gbm->device)
+    CLog::Log(LOGWARNING, "CGBMUtils::%s - device already created", __FUNCTION__);
 
-  gbm->surface = gbm_surface_create(gbm->dev,
+  gbm->device = gbm_create_device(fd);
+  if (!gbm->device)
+  {
+    CLog::Log(LOGERROR, "CGBMUtils::%s - failed to create device", __FUNCTION__);
+    return false;
+  }
+
+  return true;
+}
+
+void CGBMUtils::DestroyDevice(struct gbm *gbm)
+{
+  if (!gbm->device)
+    CLog::Log(LOGWARNING, "CGBMUtils::%s - device already destroyed", __FUNCTION__);
+
+  if (gbm->device)
+  {
+    gbm_device_destroy(gbm->device);
+    gbm->device = nullptr;
+  }
+}
+
+bool CGBMUtils::CreateSurface(struct gbm *gbm, int width, int height)
+{
+  if (gbm->surface)
+    CLog::Log(LOGWARNING, "CGBMUtils::%s - surface already created", __FUNCTION__);
```

Comment:
Same here, not sure what's the right thing to do, but this could result in leaks.

Reply:
will remove

Reply:
Sorry, this is misleading the "Same here" applies to the comment that's more above.
I don't know if it can happen that the surface is already created and you enter this again, but wether we should early return or destroy the surface before it is created a second time.

Reply:
Please keep logging for these kind of bad behaviors, a call to this method in this state should never happen, early return or a destroy call without logging will only hide a real issue.